### PR TITLE
Automation tests

### PR DESCRIPTION
**Pull request by mcgallan**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/automation/Azure.ResourceManager.Automation/tests/AutomationManagementTestBase.cs [hunk 1]
```diff
@@ -13,6 +13,9 @@ namespace Azure.ResourceManager.Automation.Tests
     public class AutomationManagementTestBase : ManagementRecordedTestBase<AutomationManagementTestEnvironment>
     {
         protected ArmClient Client { get; private set; }
+        protected AzureLocation DefaultLocation => AzureLocation.EastUS;
```
Comment (ArthurMa1978) [inline sdk/automation/Azure.ResourceManager.Automation/tests/AutomationManagementTestBase.cs:16]: The default location should be `TestEnvironment.Location`

File: sdk/automation/Azure.ResourceManager.Automation/tests/AutomationManagementTestBase.cs [hunk 2]
```diff
@@ -13,6 +13,9 @@ namespace Azure.ResourceManager.Automation.Tests
...
+        protected string groupName;
```
Comment (ArthurMa1978) [inline sdk/automation/Azure.ResourceManager.Automation/tests/AutomationManagementTestBase.cs:17]: Upcase first letter

Comment (mcgallan) [reply sdk/automation/Azure.ResourceManager.Automation/tests/AutomationManagementTestBase.cs:17]: fixed

File: sdk/automation/Azure.ResourceManager.Automation/tests/AutomationManagementTestBase.cs (hunk 1 above)
Comment (mcgallan) [reply sdk/automation/Azure.ResourceManager.Automation/tests/AutomationManagementTestBase.cs]: fixed

File: sdk/automation/Azure.ResourceManager.Automation/tests/AutomationManagementTestBase.cs [hunk 3]
```diff
@@ -25,17 +28,37 @@ protected AutomationManagementTestBase(bool isAsync)
...
+                WaitUntil.Completed,
+                resourceGroupName,
+                new ResourceGroupData(DefaultLocation)
+                {
+                    Tags =
+                    {
+                        { "test", "env" }
+                    }
+                });
+            if (Mode == RecordedTestMode.Playback)
```
Comment (ArthurMa1978) [inline sdk/automation/Azure.ResourceManager.Automation/tests/AutomationManagementTestBase.cs:50]: Please refer the WebPubSub case in https://github.com/Azure/azure-sdk-for-net/pull/35831 which not exclude recording for Network related operations but filter out the version of Network APIs

File: sdk/automation/Azure.ResourceManager.Automation/tests/Helpers/ResourceDataHelpers.cs [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+            {
+                RepoUri = new Uri("https://dev.azure.com/vinkumar0563/_git/VinKumar-AzureAutomation"),
+                Branch = "sdktest",
+                FolderPath = "/Runbooks/PowershellScripts",
+                IsAutoSyncEnabled = false,
+                IsAutoPublishRunbookEnabled = true,
+                SourceType = SourceControlSourceType.VsoGit,
+                SecurityToken = new SourceControlSecurityTokenProperties()
+                {
+                    AccessToken = "p26mwl3frnfa4l5i6a7zjfog7k75qeac7otyfa76q3ceajmrnjoq",
```
Comment (ArthurMa1978) [inline sdk/automation/Azure.ResourceManager.Automation/tests/Helpers/ResourceDataHelpers.cs:146]: this is a key leak, please sanitize this

File: sdk/automation/Azure.ResourceManager.Automation/tests/AutomationManagementTestBase.cs (hunk 3 above)
Comment (ArthurMa1978) [reply sdk/automation/Azure.ResourceManager.Automation/tests/AutomationManagementTestBase.cs:50]: Change after that  PR got merged

Comment (ArthurMa1978) [reply sdk/automation/Azure.ResourceManager.Automation/tests/AutomationManagementTestBase.cs:50]: The pr was merged

Comment (mcgallan) [reply sdk/automation/Azure.ResourceManager.Automation/tests/AutomationManagementTestBase.cs]: fixed

File: sdk/automation/Azure.ResourceManager.Automation/tests/Helpers/ResourceDataHelpers.cs (hunk 4 above)
Comment (mcgallan) [reply sdk/automation/Azure.ResourceManager.Automation/tests/Helpers/ResourceDataHelpers.cs]: fixed

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**